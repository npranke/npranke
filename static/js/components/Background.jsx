import React from 'react'

import Landscape from '@img/background-landscape.jpg'
import Portrait from '@img/background-portrait.jpg'

function Background() {
    return (
        <picture>
            <source srcSet={ Portrait } media="(orientation: portrait)" />
            <img
                src={ Landscape }
                className="background"
                alt="Conifer in Colorado landscape"
            />
        </picture>
    )
}

export default Background
