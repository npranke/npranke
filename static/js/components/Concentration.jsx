import React from 'react'

import { getLocationPageTitle } from '../utils'
import Picture0 from '../../img/concentration-22332-pink-plateau.jpg'
import Picture1 from '../../img/concentration-22512-blue-center.jpg'
import Picture2 from '../../img/concentration-22585-teal-ripples.jpg'
import Picture3 from '../../img/concentration-22594-burgundy-spot.jpg'
import Picture4 from '../../img/concentration-23056-mauve-mound.jpg'
import Picture5 from '../../img/concentration-23080-lavendar-corner.jpg'
import Picture6 from '../../img/concentration-23103-light-pool.jpg'
import Picture7 from '../../img/concentration-23183-brown-variations.jpg'
import Picture8 from '../../img/concentration-23184-purple-surface.jpg'
import Picture9 from '../../img/concentration-23231-bright-swoop.jpg'
import Picture10 from '../../img/concentration-23233-dark-ridges.jpg'
import Picture11 from '../../img/concentration-23239-neon-plains.jpg'

class Concentration extends React.Component {
    constructor(props) {
        super(props)
        this.state = {
            isPortrait: window.matchMedia(
                '(orientation: portrait)',
            ).matches,
        }

        this.updateIsPortrait = this.updateIsPortrait.bind(this)
    }

    componentDidMount() {
        document.title = getLocationPageTitle('concentration')

        window.matchMedia('(orientation: portrait)').addEventListener(
            'change',
            this.updateIsPortrait,
            false,
        )
    }

    componentWillUnmount() {
        window.matchMedia('(orientation: portrait)').removeEventListener(
            'change',
            this.updateIsPortrait,
            false,
        )
    }

    updateIsPortrait() {
        this.setState({
            isPortrait: window.matchMedia(
                '(orientation: portrait)',
            ).matches,
        })
    }

    render() {
        const board = this.state.isPortrait
            ? (
                <div className="board">
                    <div className="board-row-wrapper">
                        <div className="board-row">
                            <img
                                src={ Picture0 }
                                className="picture"
                                alt="mars"
                            />
                            <img
                                src={ Picture0 }
                                className="picture"
                                alt="mars"
                            />
                            <img
                                src={ Picture1 }
                                className="picture"
                                alt="mars"
                            />
                            <img
                                src={ Picture1 }
                                className="picture"
                                alt="mars"
                            />
                        </div>
                    </div>
                    <div className="board-row-wrapper">
                        <div className="board-row">
                            <img
                                src={ Picture2 }
                                className="picture"
                                alt="mars"
                            />
                            <img
                                src={ Picture2 }
                                className="picture"
                                alt="mars"
                            />
                            <img
                                src={ Picture3 }
                                className="picture"
                                alt="mars"
                            />
                            <img
                                src={ Picture3 }
                                className="picture"
                                alt="mars"
                            />
                        </div>
                    </div>
                    <div className="board-row-wrapper">
                        <div className="board-row">
                            <img
                                src={ Picture4 }
                                className="picture"
                                alt="mars"
                            />
                            <img
                                src={ Picture4 }
                                className="picture"
                                alt="mars"
                            />
                            <img
                                src={ Picture5 }
                                className="picture"
                                alt="mars"
                            />
                            <img
                                src={ Picture5 }
                                className="picture"
                                alt="mars"
                            />
                        </div>
                    </div>
                    <div className="board-row-wrapper">
                        <div className="board-row">
                            <img
                                src={ Picture6 }
                                className="picture"
                                alt="mars"
                            />
                            <img
                                src={ Picture6 }
                                className="picture"
                                alt="mars"
                            />
                            <img
                                src={ Picture7 }
                                className="picture"
                                alt="mars"
                            />
                            <img
                                src={ Picture7 }
                                className="picture"
                                alt="mars"
                            />
                        </div>
                    </div>
                    <div className="board-row-wrapper">
                        <div className="board-row">
                            <img
                                src={ Picture8 }
                                className="picture"
                                alt="mars"
                            />
                            <img
                                src={ Picture8 }
                                className="picture"
                                alt="mars"
                            />
                            <img
                                src={ Picture9 }
                                className="picture"
                                alt="mars"
                            />
                            <img
                                src={ Picture9 }
                                className="picture"
                                alt="mars"
                            />
                        </div>
                    </div>
                    <div className="board-row-wrapper">
                        <div className="board-row">
                            <img
                                src={ Picture10 }
                                className="picture"
                                alt="mars"
                            />
                            <img
                                src={ Picture10 }
                                className="picture"
                                alt="mars"
                            />
                            <img
                                src={ Picture11 }
                                className="picture"
                                alt="mars"
                            />
                            <img
                                src={ Picture11 }
                                className="picture"
                                alt="mars"
                            />
                        </div>
                    </div>
                </div>
            ) : (
                <div className="board">
                    <div className="board-row-wrapper">
                        <div className="board-row">
                            <img
                                src={ Picture0 }
                                className="picture"
                                alt="mars"
                            />
                            <img
                                src={ Picture0 }
                                className="picture"
                                alt="mars"
                            />
                            <img
                                src={ Picture1 }
                                className="picture"
                                alt="mars"
                            />
                            <img
                                src={ Picture1 }
                                className="picture"
                                alt="mars"
                            />
                            <img
                                src={ Picture2 }
                                className="picture"
                                alt="mars"
                            />
                            <img
                                src={ Picture2 }
                                className="picture"
                                alt="mars"
                            />
                        </div>
                    </div>
                    <div className="board-row-wrapper">
                        <div className="board-row">
                            <img
                                src={ Picture3 }
                                className="picture"
                                alt="mars"
                            />
                            <img
                                src={ Picture3 }
                                className="picture"
                                alt="mars"
                            />
                            <img
                                src={ Picture4 }
                                className="picture"
                                alt="mars"
                            />
                            <img
                                src={ Picture4 }
                                className="picture"
                                alt="mars"
                            />
                            <img
                                src={ Picture5 }
                                className="picture"
                                alt="mars"
                            />
                            <img
                                src={ Picture5 }
                                className="picture"
                                alt="mars"
                            />
                        </div>
                    </div>
                    <div className="board-row-wrapper">
                        <div className="board-row">
                            <img
                                src={ Picture6 }
                                className="picture"
                                alt="mars"
                            />
                            <img
                                src={ Picture6 }
                                className="picture"
                                alt="mars"
                            />
                            <img
                                src={ Picture7 }
                                className="picture"
                                alt="mars"
                            />
                            <img
                                src={ Picture7 }
                                className="picture"
                                alt="mars"
                            />
                            <img
                                src={ Picture8 }
                                className="picture"
                                alt="mars"
                            />
                            <img
                                src={ Picture8 }
                                className="picture"
                                alt="mars"
                            />
                        </div>
                    </div>
                    <div className="board-row-wrapper">
                        <div className="board-row">
                            <img
                                src={ Picture9 }
                                className="picture"
                                alt="mars"
                            />
                            <img
                                src={ Picture9 }
                                className="picture"
                                alt="mars"
                            />
                            <img
                                src={ Picture10 }
                                className="picture"
                                alt="mars"
                            />
                            <img
                                src={ Picture10 }
                                className="picture"
                                alt="mars"
                            />
                            <img
                                src={ Picture11 }
                                className="picture"
                                alt="mars"
                            />
                            <img
                                src={ Picture11 }
                                className="picture"
                                alt="mars"
                            />
                        </div>
                    </div>
                </div>
            )

        return (
            <main className="workbook workbook-concentration">
                <div className="concentration">
                    <div className="concentration-inner">
                        <div className="matches-wrapper">
                            <div className="matches">
                                <div className="matches-header">
                                    <div className="left-text">matches: 4</div>
                                    <div className="right-text">
                                        time: 00:00:00
                                    </div>
                                </div>
                                <div className="matches-pictures">
                                    <img
                                        src={ Picture0 }
                                        className="picture"
                                        alt="mars"
                                        id="first-match"
                                    />
                                    <img
                                        src={ Picture1 }
                                        className="picture"
                                        alt="mars"
                                    />
                                    <img
                                        src={ Picture2 }
                                        className="picture"
                                        alt="mars"
                                    />
                                    <img
                                        src={ Picture3 }
                                        className="picture"
                                        alt="mars"
                                    />
                                    <img
                                        src={ Picture4 }
                                        className="picture"
                                        alt="mars"
                                    />
                                    <img
                                        src={ Picture5 }
                                        className="picture"
                                        alt="mars"
                                    />
                                    <img
                                        src={ Picture6 }
                                        className="picture"
                                        alt="mars"
                                    />
                                    <img
                                        src={ Picture7 }
                                        className="picture"
                                        alt="mars"
                                    />
                                    <img
                                        src={ Picture8 }
                                        className="picture"
                                        alt="mars"
                                    />
                                    <img
                                        src={ Picture9 }
                                        className="picture"
                                        alt="mars"
                                    />
                                    <img
                                        src={ Picture10 }
                                        className="picture"
                                        alt="mars"
                                    />
                                    <img
                                        src={ Picture11 }
                                        className="picture"
                                        alt="mars"
                                    />
                                </div>
                            </div>
                        </div>
                        <div className="board-wrapper">
                            { board }
                        </div>
                    </div>
                </div>
            </main>
        )
    }
}

export default Concentration
