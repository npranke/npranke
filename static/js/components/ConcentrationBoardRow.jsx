import PropTypes from 'prop-types'
import React from 'react'

export function ConcentrationBoardRow(props) {
    const rowPictures = props.pictures.map((picture) => {
        const pictureName = Object.keys(picture)[0]

        const { id, pictureid } = picture[pictureName]

        let pictureImage
        if (props.first.id === id || props.second.id === id) {
            pictureImage = picture[pictureName].front
        } else if (props.matches.indexOf(pictureid) !== -1) {
            pictureImage = picture[pictureName].matched
        } else {
            pictureImage = picture[pictureName].back
        }

        return (
            <React.Fragment key={ id }>
                { pictureImage }
            </React.Fragment>
        )
    })

    return (
        <div className="board-row">
            { rowPictures }
        </div>
    )
}

ConcentrationBoardRow.defaultProps = {
    first: {
        id: null,
        pictureid: null,
    },
    matches: [],
    second: {
        id: null,
        pictureid: null,
    },
}

ConcentrationBoardRow.propTypes = {
    first: PropTypes.shape({
        id: PropTypes.string,
        pictureid: PropTypes.string,
    }),
    matches: PropTypes.arrayOf(PropTypes.string),
    pictures: PropTypes.arrayOf(
        PropTypes.shape({
            id: PropTypes.string,
            pictureid: PropTypes.string,
            back: PropTypes.node,
            front: PropTypes.node,
            matched: PropTypes.node,
        }),
    ).isRequired,
    second: PropTypes.shape({
        id: PropTypes.string,
        pictureid: PropTypes.string,
    }),
}

export default ConcentrationBoardRow
