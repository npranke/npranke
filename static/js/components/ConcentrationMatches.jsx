import PropTypes from 'prop-types'
import React from 'react'

export function ConcentrationMatches(props) {
    const matchedPictures = props.matches.map((pictureid) => {
        return (
            <React.Fragment key={ pictureid }>
                { props.pictures[`picture${pictureid}`].image }
            </React.Fragment>
        )
    })

    return (
        <div className="matches">
            <div className="matches-header">
                <div className="left-text">
                    <span className="matches-portion">
                        matches:&emsp;
                    </span>
                    <span className="matches-portion number">
                        {
                            props.matches.length < 10
                                ? `0${props.matches.length}`
                                : `${props.matches.length}`
                        }
                    </span>
                </div>
                <div className="right-text">
                    <span className="time-portion">
                        time:&emsp;
                    </span>
                    <span className="time-portion number">
                        { props.minutes }
                    </span>
                    <span className="time-portion">
                        &ensp;:&ensp;
                    </span>
                    <span className="time-portion number">
                        { props.seconds }
                    </span>
                    <span className="time-portion">
                        &ensp;:&ensp;
                    </span>
                    <span className="time-portion number">
                        { props.centiseconds }
                    </span>
                </div>
            </div>
            <div className="matches-pictures">
                { matchedPictures }
            </div>
        </div>
    )
}

ConcentrationMatches.defaultProps = {
    centiseconds: '00',
    matches: [],
    minutes: '00',
    seconds: '00',
}

ConcentrationMatches.propTypes = {
    centiseconds: PropTypes.string,
    matches: PropTypes.arrayOf(PropTypes.string),
    minutes: PropTypes.string,
    pictures: PropTypes.objectOf(
        PropTypes.shape({
            image: PropTypes.node.isRequired,
        }).isRequired,
    ).isRequired,
    seconds: PropTypes.string,
}

export default ConcentrationMatches
