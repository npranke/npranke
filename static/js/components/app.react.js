import React from 'react';

class App extends React.Component {
    render() {
        return (
            <div>
                Hello frontend world, hello!!
            </div>
        );
    }
}

export default App;
