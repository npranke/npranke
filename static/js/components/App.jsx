import React from 'react'
import { Route, Switch } from 'react-router-dom'

import Background from './Background'
import Concentration from './Concentration'
import Footer from './Footer'
import Header from './Header'
import PageNotFound from './PageNotFound'
import Welcome from './Welcome'
import Workbook from './Workbook'

function App() {
    return (
        <div className="app">
            <Background />
            <Switch>
                <Route
                    exact
                    path={ [
                        '/',
                        '/home',
                        '/workbook',
                        '/workbook/concentration',
                    ] }
                    component={ Header }
                />
                <Route
                    render={
                        (props) => {
                            return <Header { ...props } isPageNotFound />
                        }
                    }
                />
            </Switch>
            <Switch>
                <Route exact path={ ['/', '/home'] } component={ Welcome } />
                <Route exact path="/workbook" component={ Workbook } />
                <Route
                    exact
                    path="/workbook/concentration"
                    component={ Concentration }
                />
                <Route component={ PageNotFound } />
            </Switch>
            <Footer />
        </div>
    )
}

export default App
