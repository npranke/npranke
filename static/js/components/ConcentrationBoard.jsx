import PropTypes from 'prop-types'
import React from 'react'

import ConcentrationBoardRow from './ConcentrationBoardRow'

export function ConcentrationBoard(props) {
    const boardPictures = props.boardOrder.map((pictureName) => {
        return ({ [pictureName]: props.pictures[pictureName] })
    })

    return (
        props.isPortrait
            ? (
                <div className="board">
                    <div className="board-row-wrapper">
                        <ConcentrationBoardRow
                            pictures={ boardPictures.slice(0, 4) }
                            matches={ props.matches }
                            first={ props.first }
                            second={ props.second }
                        />
                    </div>
                    <div className="board-row-wrapper">
                        <ConcentrationBoardRow
                            pictures={ boardPictures.slice(4, 8) }
                            matches={ props.matches }
                            first={ props.first }
                            second={ props.second }
                        />
                    </div>
                    <div className="board-row-wrapper">
                        <ConcentrationBoardRow
                            pictures={ boardPictures.slice(8, 12) }
                            matches={ props.matches }
                            first={ props.first }
                            second={ props.second }
                        />
                    </div>
                    <div className="board-row-wrapper">
                        <ConcentrationBoardRow
                            pictures={ boardPictures.slice(12, 16) }
                            matches={ props.matches }
                            first={ props.first }
                            second={ props.second }
                        />
                    </div>
                    <div className="board-row-wrapper">
                        <ConcentrationBoardRow
                            pictures={ boardPictures.slice(16, 20) }
                            matches={ props.matches }
                            first={ props.first }
                            second={ props.second }
                        />
                    </div>
                    <div className="board-row-wrapper">
                        <ConcentrationBoardRow
                            pictures={ boardPictures.slice(20, 24) }
                            matches={ props.matches }
                            first={ props.first }
                            second={ props.second }
                        />
                    </div>
                </div>
            ) : (
                <div className="board">
                    <div className="board-row-wrapper">
                        <ConcentrationBoardRow
                            pictures={ boardPictures.slice(0, 6) }
                            matches={ props.matches }
                            first={ props.first }
                            second={ props.second }
                        />
                    </div>
                    <div className="board-row-wrapper">
                        <ConcentrationBoardRow
                            pictures={ boardPictures.slice(6, 12) }
                            matches={ props.matches }
                            first={ props.first }
                            second={ props.second }
                        />
                    </div>
                    <div className="board-row-wrapper">
                        <ConcentrationBoardRow
                            pictures={ boardPictures.slice(12, 18) }
                            matches={ props.matches }
                            first={ props.first }
                            second={ props.second }
                        />
                    </div>
                    <div className="board-row-wrapper">
                        <ConcentrationBoardRow
                            pictures={ boardPictures.slice(18, 24) }
                            matches={ props.matches }
                            first={ props.first }
                            second={ props.second }
                        />
                    </div>
                </div>
            )
    )
}

ConcentrationBoard.defaultProps = {
    first: {
        id: null,
        pictureid: null,
    },
    isPortrait: false,
    matches: [],
    second: {
        id: null,
        pictureid: null,
    },
}

ConcentrationBoard.propTypes = {
    boardOrder: PropTypes.arrayOf(PropTypes.string).isRequired,
    first: PropTypes.shape({
        id: PropTypes.string,
        pictureid: PropTypes.string,
    }),
    isPortrait: PropTypes.bool,
    matches: PropTypes.arrayOf(PropTypes.string),
    pictures: PropTypes.objectOf(
        PropTypes.shape({
            id: PropTypes.string.isRequired,
            pictureid: PropTypes.string.isRequired,
            back: PropTypes.node.isRequired,
            front: PropTypes.node.isRequired,
            matched: PropTypes.node.isRequired,
        }).isRequired,
    ).isRequired,
    second: PropTypes.shape({
        id: PropTypes.string,
        pictureid: PropTypes.string,
    }),
}

export default ConcentrationBoard
