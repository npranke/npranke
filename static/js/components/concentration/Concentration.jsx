import PropTypes from 'prop-types'
import React from 'react'

import ConcentrationBoard from '@components/concentration/ConcentrationBoard'
import ConcentrationMatches from
    '@components/concentration/ConcentrationMatches'
import withPortraitListener from '@components/hoc/PortraitListener'

import { PICTURE_BACK, PICTURE_SOURCES } from '@constants/concentration'

import { getSendEventHandler, sendEvent } from '@utils'

export class Concentration extends React.Component {
    constructor(props) {
        super(props)

        this.state = {
            isTimeRunning: false,
            turns: 0,
            internalMatches: [],
            displayedMatches: [],
            first: { id: null, pictureid: null },
            second: { id: null, pictureid: null },
        }

        this.pictureClickHandler = this.pictureClickHandler.bind(this)
        this.pictureKeyUpHandler = this.pictureKeyUpHandler.bind(this)

        this.matchesPictures = Object.keys(PICTURE_SOURCES).reduce(
            (pictureAccumulator, pictureid) => {
                const mroPIA = `NASA/JPL Mars Reconnaissance Orbiter PIA${
                    PICTURE_SOURCES[pictureid].sourceid
                }`

                pictureAccumulator[`picture${pictureid}`] = {
                    image: (
                        <a
                            onClick={ getSendEventHandler(
                                'concentration',
                                'navigate',
                                `PIA${PICTURE_SOURCES[pictureid].sourceid}`,
                            ) }
                            href={
                                'https://www.jpl.nasa.gov/spaceimages/'
                                + 'details.php?id='
                                + `PIA${PICTURE_SOURCES[pictureid].sourceid}`
                            }
                            target="_blank"
                            rel="noreferrer noopener external"
                            aria-label="NASA/JPL Space Images"
                        >
                            <img
                                src={ PICTURE_SOURCES[pictureid].source }
                                className="picture"
                                alt={ mroPIA }
                                title={ mroPIA }
                            />
                        </a>
                    ),
                }

                return pictureAccumulator
            },
            {},
        )

        this.boardPictures = Object.keys(PICTURE_SOURCES).reduce(
            (pictureAccumulator, pictureid) => {
                const mroPIA = `NASA/JPL Mars Reconnaissance Orbiter PIA${
                    PICTURE_SOURCES[pictureid].sourceid
                }`
                const merPIA = 'NASA/JPL Mars Exploration Rover PIA20285'

                pictureAccumulator[`picture${pictureid}-a`] = {
                    id: `${pictureid}-a`,
                    pictureid: `${pictureid}`,
                    back: (
                        <img
                            onClick={ this.pictureClickHandler }
                            onKeyUp={ this.pictureKeyUpHandler }
                            role="button"
                            tabIndex="0"
                            data-id={ `${pictureid}-a` }
                            data-pictureid={ pictureid }
                            src={ PICTURE_BACK }
                            className="picture picture-back"
                            alt={ merPIA }
                        />
                    ),
                    front: (
                        <img
                            src={ PICTURE_SOURCES[pictureid].source }
                            className="picture picture-front"
                            alt={ mroPIA }
                        />
                    ),
                    matched: (
                        <img
                            src={ PICTURE_SOURCES[pictureid].source }
                            className="picture picture-matched"
                            alt={ mroPIA }
                        />
                    ),
                }

                pictureAccumulator[`picture${pictureid}-b`] = {
                    id: `${pictureid}-b`,
                    pictureid: `${pictureid}`,
                    back: (
                        <img
                            onClick={ this.pictureClickHandler }
                            onKeyUp={ this.pictureKeyUpHandler }
                            role="button"
                            tabIndex="0"
                            data-id={ `${pictureid}-b` }
                            data-pictureid={ pictureid }
                            src={ PICTURE_BACK }
                            className="picture picture-back"
                            alt={ merPIA }
                        />
                    ),
                    front: (
                        <img
                            src={ PICTURE_SOURCES[pictureid].source }
                            className="picture picture-front"
                            alt={ mroPIA }
                        />
                    ),
                    matched: (
                        <img
                            src={ PICTURE_SOURCES[pictureid].source }
                            className="picture picture-matched"
                            alt={ mroPIA }
                        />
                    ),
                }

                return pictureAccumulator
            },
            {},
        )

        this.boardOrder = Object.keys(
            this.boardPictures,
        ).map((pictureName) => {
            return { sortValue: Math.random(), pictureName }
        }).sort((pictureWrapperA, pictureWrapperB) => {
            return pictureWrapperA.sortValue - pictureWrapperB.sortValue
        }).map((pictureWrapper) => {
            return pictureWrapper.pictureName
        })
    }

    componentWillUnmount() {
        clearTimeout(this.pictureTimeout)
    }

    pictureClickHandler(event) {
        const { id, pictureid } = event.currentTarget.dataset

        sendEvent('concentration', 'click', 'picture')

        this.setState((state) => {
            if (state.first.id === null) {
                return {
                    first: { id, pictureid },
                    displayedMatches: state.internalMatches,
                    isTimeRunning: true,
                }
            } else if (
                state.second.id === null && state.first.pictureid === pictureid
            ) {
                sendEvent('concentration', 'find', 'match')

                const isLastMatch = state.internalMatches.length === 11

                if (isLastMatch) {
                    sendEvent(
                        'concentration',
                        'complete',
                        'matches',
                        state.turns + 1,
                    )
                }

                this.pictureTimeout = setTimeout(() => {
                    this.setState({
                        first: { id: null, pictureid: null },
                        second: { id: null, pictureid: null },
                        displayedMatches: state.displayedMatches.concat(
                            [pictureid],
                        ),
                        isTimeRunning: !isLastMatch,
                    })
                }, 1000)

                return {
                    turns: state.turns + 1,
                    second: { id, pictureid },
                    internalMatches: state.internalMatches.concat([pictureid]),
                    displayedMatches: state.internalMatches,
                    isTimeRunning: !isLastMatch,
                }
            } else if (
                state.second.id === null && state.first.pictureid !== pictureid
            ) {
                this.pictureTimeout = setTimeout(() => {
                    this.setState({
                        first: { id: null, pictureid: null },
                        second: { id: null, pictureid: null },
                        displayedMatches: state.internalMatches,
                        isTimeRunning: true,
                    })
                }, 2000)

                return {
                    turns: state.turns + 1,
                    second: { id, pictureid },
                    displayedMatches: state.internalMatches,
                    isTimeRunning: true,
                }
            } else if (state.first.id && state.second.id) {
                clearTimeout(this.pictureTimeout)

                return {
                    first: { id, pictureid },
                    second: { id: null, pictureid: null },
                    displayedMatches: state.internalMatches,
                    isTimeRunning: true,
                }
            }

            return {
                displayedMatches: state.internalMatches,
                isTimeRunning: true,
            }
        })
    }

    pictureKeyUpHandler(event) {
        if (event.key === 'Enter' || event.key === ' ') {
            this.pictureClickHandler(event)
        }
    }

    render() {
        return (
            <div className="concentration">
                <div className="matches-wrapper">
                    <ConcentrationMatches
                        isTimeRunning={ this.state.isTimeRunning }
                        pictures={ this.matchesPictures }
                        matches={ this.state.displayedMatches }
                        turns={ this.state.turns }
                    />
                </div>
                <div className="board-wrapper">
                    <ConcentrationBoard
                        pictures={ this.boardPictures }
                        boardOrder={ this.boardOrder }
                        isPortrait={ this.props.isPortrait }
                        first={ this.state.first }
                        second={ this.state.second }
                        matches={ this.state.internalMatches }
                    />
                </div>
            </div>
        )
    }
}

Concentration.defaultProps = {
    isPortrait: false,
}

Concentration.propTypes = {
    isPortrait: PropTypes.bool,
}

export default withPortraitListener(Concentration)
