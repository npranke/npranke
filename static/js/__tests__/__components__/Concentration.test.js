import Adapter from 'enzyme-adapter-react-16'
import Enzyme, { shallow } from 'enzyme'
import React from 'react'

import { Concentration } from '../../components/Concentration'

Enzyme.configure({ adapter: new Adapter() })

describe('Concentration', () => {
    test('has matches section', () => {
        const concentration = shallow(<Concentration />)

        expect(
            concentration.find(
                '.concentration-inner .matches-wrapper .matches',
            ).exists(),
        ).toBe(true)
    })

    test('has board section', () => {
        const concentration = shallow(<Concentration />)

        expect(
            concentration.find(
                '.concentration-inner .board-wrapper .board',
            ).exists(),
        ).toBe(true)
    })
})

describe('Concentration snapshot', () => {
    test('matches snapshot', () => {
        const concentration = shallow(<Concentration />)

        expect(concentration).toMatchSnapshot()
    })
})
