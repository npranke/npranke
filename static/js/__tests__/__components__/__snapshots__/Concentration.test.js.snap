// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Concentration snapshot matches snapshot 1`] = `
<main
  className="workbook workbook-concentration"
>
  <div
    className="concentration"
  >
    <div
      className="concentration-inner"
    >
      <div
        className="matches-wrapper"
      >
        <div
          className="matches"
        >
          <div
            className="matches-header"
          >
            <div
              className="left-text"
            >
              matches: 4
            </div>
            <div
              className="right-text"
            >
              time: 00:00:00
            </div>
          </div>
          <div
            className="matches-pictures"
          >
            <img
              alt="mars"
              className="picture"
              id="first-match"
              src="image-mock"
            />
            <img
              alt="mars"
              className="picture"
              src="image-mock"
            />
            <img
              alt="mars"
              className="picture"
              src="image-mock"
            />
            <img
              alt="mars"
              className="picture"
              src="image-mock"
            />
            <img
              alt="mars"
              className="picture"
              src="image-mock"
            />
            <img
              alt="mars"
              className="picture"
              src="image-mock"
            />
            <img
              alt="mars"
              className="picture"
              src="image-mock"
            />
            <img
              alt="mars"
              className="picture"
              src="image-mock"
            />
            <img
              alt="mars"
              className="picture"
              src="image-mock"
            />
            <img
              alt="mars"
              className="picture"
              src="image-mock"
            />
            <img
              alt="mars"
              className="picture"
              src="image-mock"
            />
            <img
              alt="mars"
              className="picture"
              src="image-mock"
            />
          </div>
        </div>
      </div>
      <div
        className="board-wrapper"
      >
        <div
          className="board"
        >
          <div
            className="board-row-wrapper"
          >
            <div
              className="board-row"
            >
              <img
                alt="mars"
                className="picture"
                src="image-mock"
              />
              <img
                alt="mars"
                className="picture"
                src="image-mock"
              />
              <img
                alt="mars"
                className="picture"
                src="image-mock"
              />
              <img
                alt="mars"
                className="picture"
                src="image-mock"
              />
              <img
                alt="mars"
                className="picture"
                src="image-mock"
              />
              <img
                alt="mars"
                className="picture"
                src="image-mock"
              />
            </div>
          </div>
          <div
            className="board-row-wrapper"
          >
            <div
              className="board-row"
            >
              <img
                alt="mars"
                className="picture"
                src="image-mock"
              />
              <img
                alt="mars"
                className="picture"
                src="image-mock"
              />
              <img
                alt="mars"
                className="picture"
                src="image-mock"
              />
              <img
                alt="mars"
                className="picture"
                src="image-mock"
              />
              <img
                alt="mars"
                className="picture"
                src="image-mock"
              />
              <img
                alt="mars"
                className="picture"
                src="image-mock"
              />
            </div>
          </div>
          <div
            className="board-row-wrapper"
          >
            <div
              className="board-row"
            >
              <img
                alt="mars"
                className="picture"
                src="image-mock"
              />
              <img
                alt="mars"
                className="picture"
                src="image-mock"
              />
              <img
                alt="mars"
                className="picture"
                src="image-mock"
              />
              <img
                alt="mars"
                className="picture"
                src="image-mock"
              />
              <img
                alt="mars"
                className="picture"
                src="image-mock"
              />
              <img
                alt="mars"
                className="picture"
                src="image-mock"
              />
            </div>
          </div>
          <div
            className="board-row-wrapper"
          >
            <div
              className="board-row"
            >
              <img
                alt="mars"
                className="picture"
                src="image-mock"
              />
              <img
                alt="mars"
                className="picture"
                src="image-mock"
              />
              <img
                alt="mars"
                className="picture"
                src="image-mock"
              />
              <img
                alt="mars"
                className="picture"
                src="image-mock"
              />
              <img
                alt="mars"
                className="picture"
                src="image-mock"
              />
              <img
                alt="mars"
                className="picture"
                src="image-mock"
              />
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</main>
`;
